app/core/optimizations.py

```
@@ -0,0 +1,115 @@
+    async def _record_metrics(self, operation_name: str, execution_time: float):
+        """Record performance metrics for an operation."""
+        try:
+            self._metrics.execution_time += execution_time
+            # Submit memory usage tracking to thread pool
+            self._executor.submit(self._track_memory_usage)
+            logger.info(f"Operation {operation_name} completed in {execution_time:.3f}s")
+        except Exception as e:
+            logger.error(f"Error recording metrics: {e}")
+            self._metrics.error_count += 1
+            
+    def _track_memory_usage(self):
```

Ruff (0.8.2)

Missing return type annotation for private function `_track_memory_usage`

Add return type annotation: `None`

(ANN202)